rpcserver.go

```diff
@@ -3027,6 +3027,88 @@ func (s *rpcServer) determineAccountVersion(
 	return account.VersionTaprootEnabled, nil
 }
 
+// AccountModificationFees returns a map from account key to an ordered list of
+// account modifying action fees.
+func (s *rpcServer) AccountModificationFees(ctx context.Context,
+	req *poolrpc.AccountModificationFeesRequest) (
+	*poolrpc.AccountModificationFeesResponse, error) {
+	// Retrieve all lightning wallet transactions.
+	transactionDetails, err := s.lndClient.GetTransactions(
+		ctx, &lnrpc.GetTransactionsRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving transactions: %v", err)
+	}
+
+	// A map from account key to an ordered list of account action fees.
+	accountActions := make(map[string][]*poolrpc.AccountModificationFee)
+
+	for _, tx := range transactionDetails.Transactions {
+		// Skip transactions which are
+		if !account.IsPoolTx(tx) {
+			continue
+		}
+
+		// Parse account data from transaction label.
+		labelData, err := account.ParseTxLabel(tx.Label)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse account "+
+				"transaction label: %v (error: %v)", tx.Label, err)
+		}
+		// Ignore transaction if account modification not found.
+		if labelData == nil {
+			continue
+		}
+
+		// Select the account specific transaction output. Each account
+		// action transaction should include an account specific output.
+		output := account.SelectOutput(tx, labelData.OutputIndex)
+		if output != nil {
+			return nil, fmt.Errorf("failed to identify account "+
+				"tx output: %v", tx)
+		}
+
+		fee := btcutil.Amount(0)
+		if labelData.TxFee != nil {
+			fee = *labelData.TxFee
+		}
```

Comment:
This is a bit ugly. Technically the fee could be missing from the tx label json (if lnd couldn't be reached). Is there a better way to do this?